contracts/scripts/test.sh

```diff
@@ -4,32 +4,21 @@ set -euo pipefail
 
 E2E_PATH=$(pwd)/e2e-tests
 CONTRACTS_PATH=$(pwd)/contracts
-EARLY_BIRD_SPECIAL=$(jq --raw-output ".early_bird_special" < "$CONTRACTS_PATH"/addresses.json)
-THE_PRESSIAH_COMETH=$(jq --raw-output ".the_pressiah_cometh" < "$CONTRACTS_PATH"/addresses.json)
-BACK_TO_THE_FUTURE=$(jq --raw-output ".back_to_the_future" < "$CONTRACTS_PATH"/addresses.json)
-SIMPLE_DEX=$(jq --raw-output ".simple_dex" < "$CONTRACTS_PATH"/addresses.json)
-WRAPPED_AZERO=$(jq --raw-output ".wrapped_azero" < "$CONTRACTS_PATH"/addresses.json)
 
 pushd "$E2E_PATH"
 
-RUST_LOG="aleph_e2e_client=info" cargo run --release -- \
-  --test-cases wrapped_azero \
-  --test-cases simple_dex \
-  --test-cases marketplace \
-  --test-cases button_game_reset \
-  --test-cases early_bird_special \
-  --test-cases the_pressiah_cometh \
-  --test-cases back_to_the_future \
-  --early-bird-special "$EARLY_BIRD_SPECIAL" \
-  --the-pressiah-cometh "$THE_PRESSIAH_COMETH" \
-  --back-to-the-future "$BACK_TO_THE_FUTURE" \
-  --simple-dex "$SIMPLE_DEX" \
-  --wrapped-azero "$WRAPPED_AZERO" \
-  --button-game-metadata ../contracts/button/target/ink/metadata.json \
-  --ticket-token-metadata ../contracts/ticket_token/target/ink/metadata.json \
-  --reward-token-metadata ../contracts/game_token/target/ink/metadata.json \
-  --marketplace-metadata ../contracts/marketplace/target/ink/metadata.json \
-  --simple-dex-metadata ../contracts/simple_dex/target/ink/metadata.json \
-  --wrapped-azero-metadata ../contracts/wrapped_azero/target/ink/metadata.json
+EARLY_BIRD_SPECIAL=$(jq --raw-output ".early_bird_special" < "$CONTRACTS_PATH"/addresses.json) \
+  THE_PRESSIAH_COMETH=$(jq --raw-output ".the_pressiah_cometh" < "$CONTRACTS_PATH"/addresses.json) \
+  BACK_TO_THE_FUTURE=$(jq --raw-output ".back_to_the_future" < "$CONTRACTS_PATH"/addresses.json) \
+  SIMPLE_DEX=$(jq --raw-output ".simple_dex" < "$CONTRACTS_PATH"/addresses.json) \
+  WRAPPED_AZERO=$(jq --raw-output ".wrapped_azero" < "$CONTRACTS_PATH"/addresses.json) \
+  BUTTON_GAME_METADATA=$CONTRACTS_PATH/button/target/ink/metadata.json \
+  TICKET_TOKEN_METADATA=$CONTRACTS_PATH/ticket_token/target/ink/metadata.json \
+  REWARD_TOKEN_METADATA=$CONTRACTS_PATH/game_token/target/ink/metadata.json \
+  MARKETPLACE_METADATA=$CONTRACTS_PATH/marketplace/target/ink/metadata.json \
+  SIMPLE_DEX_METADATA=$CONTRACTS_PATH/simple_dex/target/ink/metadata.json \
+  WRAPPED_AZERO_METADATA=$CONTRACTS_PATH/wrapped_azero/target/ink/metadata.json \
+  RUST_LOG="aleph_e2e_client=info" \
+  cargo test button -- --test-threads 1 --nocapture
```

Comment:
I think we need to drop the `button` and go for the 
```suggestion
  cargo test -- --test-threads 1 --nocapture
```
otherwise we won't test `simple_dex` at the very least.

Reply:
```
yapee@skadi:~/Projects/aleph-node/e2e-tests$ cargo test button -- --list 

test::button_game::back_to_the_future: test
test::button_game::button_game_reset: test
test::button_game::early_bird_special: test
test::button_game::marketplace: test
test::button_game::simple_dex: test
test::button_game::the_pressiah_cometh: test
test::button_game::wrapped_azero: test
```

I don't think so - the filter is applied to the whole test name, and all of these are in `button_game`, so they are all included.

Reply:
Oh, right. I forgot there's `button_game` as part of the test path.